jdbcSource/src/main/java/io/vantiq/extsrc/jdbcSource/JDBCCore.java

```diff
@@ -192,17 +193,42 @@ public void executeQuery(ExtensionServiceMessage message) {
                         "JDBC connection closed before operation could complete.", null);
             }
         }
-        
+
         // Gather query results and send the appropriate response, or send a query error if an exception is caught
         try {
             if (request.get("query") instanceof String) {
                 String queryString = (String) request.get("query");
-                HashMap[] queryArray = localJDBC.processQuery(queryString);
-                sendDataFromQuery(queryArray, message);
+                // Check if SQL Query is an update statement, or query statement
+                if (!queryString.trim().toLowerCase().startsWith(SELECT_STATEMENT_IDENTIFIER)) {
```

Comment:
A style/readability tip:
if you have both IF and and ELSE clauses, I think it's more readable if you don't put the NOT in the test.
E.g.  I would reverse this IF and its clauses:  
     if (queryString.trim().toLowerCase().startsWith(SELECT_STATEMENT_IDENTIFIER)) ...

Other things I try to do:  
-I try to put the "common case" into the IF clause and the "exception case" into the ELSE clause...
-When one clause has 2-3 lines and the other has more lines, I try to put the smaller/simpler clause in  the IF half, so you can quickly see that & then move on to reading the ELSE clause.

Reply:
Thanks for the advice Roger! I'll fix this later this afternoon.